tests/unit/test_workflow_files.py

```diff
@@ -301,13 +301,32 @@ def test_infer_latest_run(
         assert infer_latest_run(path) == expected
 
 
+def test_infer_latest_run_raise_warning_for_runN(caplog, tmp_run_dir):
+    caplog.set_level(logging.WARNING, 'log')
+    cylc_run_dir: Path = tmp_run_dir()
+    run_dir = cylc_run_dir / 'run1'
+    run_dir.mkdir()
+    runN_path = cylc_run_dir / 'runN'
+    runN_path.symlink_to('run1')
+    infer_latest_run(runN_path)
+    assert ("The use of runN in the Workflow ID is not necessary."
```

Comment:
Consider using a regex or similar to check that the essentials of this message are there, without making it very fragile in terms of exact words used?